src/CommandRunner.cpp

```diff
@@ -0,0 +1,204 @@
+#include "CommandRunner.hpp"
+#include <memory>
+#include <stdio.h>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <time.h>
+
+#include <glog/logging.h>
+#include <stout/try.hpp>
+
+#define TEMP_FILE_TEMPLATE "/tmp/criteo-mesos-XXXXXX"
+
+#define READ 0
+#define WRITE 1
+
+namespace criteo {
+namespace mesos {
+namespace CommandRunner {
+
+/*
+ * Represent a temporary file that can be either written or read from.
+ */
+class TemporaryFile
+{
+public:
+  TemporaryFile() {
+    char filepath[] = TEMP_FILE_TEMPLATE;
+    if(mkstemp(filepath) == -1)
+      throw std::runtime_error(
+        "Unable to create temporary file to run commands");
+    m_filepath = std::string(filepath);
+  }
+
+  /*
+   * Read whole content of the temporary file.
+   * @return The content of the file.
+   */
+  std::string readAll() const {
+    std::ifstream ifs(m_filepath);
+    std::string content((std::istreambuf_iterator<char>(ifs)),
+                        (std::istreambuf_iterator<char>()));
+    ifs.close();
+    return content;
+  }
+
+  /*
+   * Write content to the temporary file and flush it.
+   * @param content The content to write to the file.
+   */
+  void write(const std::string& content) const {
+    std::ofstream ofs;
+    ofs.open(m_filepath);
+    ofs << content;
+    std::flush(ofs);
+    ofs.close();
+  }
+
+  ~TemporaryFile() {
+    if(remove(m_filepath.c_str()) != 0)
+      std::cerr << "Error while deleting " << m_filepath << std::endl;
+  }
+
+  inline const std::string& filepath() const { return m_filepath; }
+
+private:
+  std::string m_filepath;
+};
+
+
+/*
+ * Fork and run a command in the child process.
+ *
+ * This implementation has been inspired by
+ * https://dzone.com/articles/simple-popen2-implementation
+ *
+ * @param command The command to run in the child process.
+ * @param infp The input FILE pointer to receive data from child.
+ * @param outfp The output FILE pointer to send data to child.
+ * @return The pid of the child process.
+ */
+pid_t popen2(const char *command, int *infp = NULL, int *outfp = NULL)
+{
+  int p_stdin[2], p_stdout[2];
+  pid_t pid;
+  if (pipe(p_stdin) != 0 || pipe(p_stdout) != 0)
+    return -1;
```

Comment:
Add brackets to prevent futur refactor to fails

Reply:
Will do